cluster-commands.go

```diff
@@ -0,0 +1,357 @@
+type BktOp string
+
+// BktOp value constants.
+const (
+	// make bucket and enable versioning
+	MakeWithVersioningBktOp BktOp = "make-with-versioning"
+	// add replication configuration
+	ConfigureReplBktOp BktOp = "configure-replication"
+	// delete bucket (forceDelete = off)
+	DeleteBucketBktOp BktOp = "delete-bucket"
+	// delete bucket (forceDelete = on)
+	ForceDeleteBucketBktOp BktOp = "force-delete-bucket"
+)
+
+// CRInternalBucketOps - tells peers to create bucket and setup replication.
+func (adm *AdminClient) CRInternalBucketOps(ctx context.Context, bucket string, op BktOp, opts map[string]string) error {
+	v := url.Values{}
+	v.Add("bucket", bucket)
+	v.Add("operation", string(op))
+
+	// For make-bucket, bucket options may be sent via `opts`
+	if op == MakeWithVersioningBktOp {
+		for k, val := range opts {
+			v.Add(k, val)
+		}
+	}
+	reqData := requestData{
+		queryValues: v,
+		relPath:     adminAPIPrefix + "/site-replication/peer/bucket-ops",
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	return nil
+}
+
+// CRIAMItem.Type constants.
+const (
+	CRIAMItemPolicy        = "policy"
+	CRIAMItemSvcAcc        = "service-account"
+	CRIAMItemPolicyMapping = "policy-mapping"
+)
+
+// CRSvcAccCreate - create operation
+type CRSvcAccCreate struct {
+	Parent        string            `json:"parent"`
+	AccessKey     string            `json:"accessKey"`
+	SecretKey     string            `json:"secretKey"`
+	Groups        []string          `json:"groups"`
+	LDAPUser      string            `json:"ldapUser"`
+	SessionPolicy *iampolicy.Policy `json:"sessionPolicy"`
+	Status        string            `json:"status"`
+}
+
+// CRSvcAccUpdate - update operation
+type CRSvcAccUpdate struct {
+	AccessKey     string            `json:"accessKey"`
+	SecretKey     string            `json:"secretKey"`
+	Status        string            `json:"status"`
+	SessionPolicy *iampolicy.Policy `json:"sessionPolicy"`
+}
+
+// CRSvcAccDelete - delete operation
+type CRSvcAccDelete struct {
+	AccessKey string `json:"accessKey"`
+}
+
+// CRSvcAccChange - sum-type to represent an svc account change.
+type CRSvcAccChange struct {
+	Create *CRSvcAccCreate `json:"crSvcAccCreate"`
+	Update *CRSvcAccUpdate `json:"crSvcAccUpdate"`
+	Delete *CRSvcAccDelete `json:"crSvcAccDelete"`
+}
+
+// CRPolicyMapping - represents mapping of a policy to a user or group.
+type CRPolicyMapping struct {
+	UserOrGroup string `json:"userOrGroup"`
+	IsGroup     bool   `json:"isGroup"`
+	Policy      string `json:"policy"`
+}
+
+// CRIAMItem - represents an IAM object that will be copied to a peer.
+type CRIAMItem struct {
+	Type string `json:"type"`
+
+	// Name and Policy below are used when Type == CRIAMItemPolicy
+	Name   string            `json:"name"`
+	Policy *iampolicy.Policy `json:"policy"`
```

Comment:
json.RawMessage is appropriate here @donatello

Reply:
> json.RawMessage is appropriate here @donatello

👍🏾